table/block_based/block_based_table_reader.h

```diff
@@ -568,7 +571,8 @@ class BlockBasedTableIterator : public InternalIteratorBase<TValue> {
                           const SliceTransform* prefix_extractor,
                           BlockType block_type, bool key_includes_seq = true,
                           bool index_key_is_full = true,
-                          bool for_compaction = false)
+                          bool for_compaction = false,
+                          size_t compaction_readahead_size = 0)
```

Comment:
Instead of passing in compaction_readahead_size, it might be more generic to pass in EnvOptions from table_cache until here, and in BlockBasedTableIterator constructor set the correct readahead_size based on if this is a user read or compaction read.

Reply:
Discussed offline will leave it as is for now. compaction_readahead_size is a natural extension to previously added flag for_compaction.